data-service/src/test/java/com/amazonaws/blox/dataservice/repository/ListClustersTest.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://aws.amazon.com/apache2.0/
+ *
+ * or in the "LICENSE" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package com.amazonaws.blox.dataservice.repository;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.assertj.core.groups.Tuple.tuple;
+
+import com.amazonaws.blox.dataservice.model.EnvironmentId;
+import com.amazonaws.blox.dataservice.test.data.ModelBuilders;
+import java.util.Arrays;
+import java.util.List;
+import org.junit.Test;
+
+public class ListClustersTest extends EnvironmentRepositoryTestCase {
+  private static final String CLUSTER = "ClusterOne";
+  private static final String OTHER_CLUSTER = "ClusterTwo";
+  private static final String ENVIRONMENT_NAME = "EnvironmentOne";
+  private static final String OTHER_ENVIRONMENT_NAME = "EnvironmentTwo";
+  ModelBuilders models = ModelBuilders.builder().build();
+
+  List<EnvironmentId> ids =
+      Arrays.asList(
+          models.environmentId("1", "ClusterOne", "EnvironmentOne"),
+          models.environmentId("2", "ClusterTwo", "EnvironmentTwo"),
+          models.environmentId("2", "ClusterThree", "EnvironmentThree"));
+
+  public void givenEnvironments(List<EnvironmentId> ids) throws Exception {
+    for (EnvironmentId id : ids) {
+      repo.createEnvironmentAndEnvironmentRevision(
+          models.environment().environmentId(id).build(),
+          models.environmentRevision().environmentId(id).build());
+    }
+  }
+
+  public void givenEnvironments(EnvironmentId... ids) throws Exception {
+    givenEnvironments(Arrays.asList(ids));
+  }
+
+  @Test
+  public void itDoesntIncludeDuplicateClusters() throws Exception {
+    givenEnvironments(ids);
```

Comment:
Looks `givenEnvironments(ids);` is used for all test cases, maybe move into a @Before

Reply:
I had it that way before, and then I had to use slightly different givens for each case, and it was weird that some givens were in the before block and some were in the test case. I think this is starting to smell of the Large Shared Fixture antipattern, let me try your suggestion below instead.